client/app/lib/statefulForm.js

```diff
@@ -0,0 +1,127 @@
+/** @jsx h */
+import { h, Component } from 'preact'
```

Comment:
What do you think to use the `.jsx` extension here? Even if it's more an helper than a component.